src/Tribe/RSVP.php

```diff
@@ -922,139 +883,108 @@ public function create_attendee_for_ticket( $ticket, $attendee_data ) {
-		$rsvp_order_ids = array_map( 'absint', wp_list_pluck( $rsvp_orders, 'order_id' ) );
+		$rsvp_attendees    = $this->tickets_view->get_event_rsvp_attendees( $post_id, $user_id );
+		$rsvp_attendee_ids = array_map( 'absint', wp_list_pluck( $rsvp_attendees, 'attendee_id' ) );
 
 		// This makes sure we don't save attendees for orders that are not from this current user and event
-		if ( ! in_array( (int) $order_id, $rsvp_order_ids, true ) ) {
+		if ( ! in_array( $attendee_id, $rsvp_attendee_ids, true ) ) {
 			return;
 		}
 
 		$attendee = array();
 
-		// Get the Attendee Data, it's important for testing
-		foreach ( $rsvp_orders as $test_attendee ) {
-			if ( $order_id !== $test_attendee['order_id'] ) {
+		// Get the attendee data for the attendee we are updating.
+		foreach ( $rsvp_attendees as $test_attendee ) {
+			if ( $attendee_id !== $test_attendee['attendee_id'] ) {
 				continue;
 			}
 
 			$attendee = $test_attendee;
 		}
 
 		// Don't try to Save if it's restricted
-		if ( ! isset( $attendee['product_id'] )
-		     || $this->tickets_view->is_rsvp_restricted( $event_id, $attendee['product_id'] )
+		if (
+			! isset( $attendee['product_id'] )
+		    || $this->tickets_view->is_rsvp_restricted( $post_id, $attendee['product_id'] )
 		) {
 			return;
 		}
 
-		$attendee_email     = empty( $data['email'] ) ? null : sanitize_email( $data['email'] );
-		$attendee_email     = is_email( $attendee_email ) ? $attendee_email : null;
-		$attendee_full_name = empty( $data['full_name'] ) ? null : sanitize_text_field( $data['full_name'] );
-		$attendee_optout    = empty( $data['optout'] ) ? 0 : $data['optout'];
+		$attendee_data_to_save = [];
 
-		$attendee_optout = filter_var( $attendee_optout, FILTER_VALIDATE_BOOLEAN );
-		$attendee_optout = (int) $attendee_optout;
+		// Only update full name if set.
+		if ( ! empty( $attendee_data['full_name'] ) ) {
+			$attendee_data_to_save['full_name'] = sanitize_text_field( $attendee_data['full_name'] );
+		}
+
+		// Only update email if set.
+		if ( ! empty( $attendee_data['email'] ) ) {
+			$attendee_data['email'] = sanitize_email( $attendee_data['email'] );
+
+			// Only update email if valid
+			if ( is_email( $attendee_data['email'] ) ) {
+				$attendee_data_to_save['email'] = $attendee_data['email'];
+			}
+		}
 
-		if ( empty( $data['order_status'] ) || ! $this->tickets_view->is_valid_rsvp_option( $data['order_status'] ) ) {
-			$attendee_order_status = null;
+		// Only update optout if set.
+		if ( isset( $attendee_data['optout'] ) ) {
+			$attendee_data_to_save['optout'] = (int) tribe_is_truthy( $attendee_data['optout'] );
+		}
+
+		if ( empty( $attendee_data['order_status'] ) || ! $this->tickets_view->is_valid_rsvp_option( $attendee_data['order_status'] ) ) {
+			$attendee_status = null;
 		} else {
-			$attendee_order_status = $data['order_status'];
+			$attendee_status = $attendee_data['order_status'];
 		}
 
-		$product_id  = $attendee['product_id'];
+		$ticket_id  = $attendee['product_id'];
```

Comment:
:warning: Equals sign not aligned correctly; expected 1 space but found 2 spaces

<sub>hash: 3d840d5bb61ed2fe521df1f95a431215</sub>